plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -329,10 +330,16 @@ export class ElasticSearchSearchEngine implements SearchEngine {
         nextPageCursor,
         previousPageCursor,
       };
-    } catch (e) {
+    } catch (error) {
+      if (error.meta.body.error.type === 'index_not_found_exception') {
+        throw new MissingIndexError(
+          `Missing index for ${queryIndices}. This means there are no documents to search through.`,
+          error,
+        );
+      }
       this.logger.error(
         `Failed to query documents for indices ${queryIndices}`,
-        e,
+        error,
       );
       return Promise.reject({ results: [] });
```

Comment:
I know it's not part of the PR, but did you really intend to do a rejection here rather than a resolve? Rejections should be with an Error (otherwise an `await` will throw this object which is kinda unexpected)

Reply:
Good point, I don't have the full context on why it was implemented this way in the first place. But we should look into this separately.